### PR TITLE
fix(trainer_distill): address code review feedback on distillation trainer and validator

### PR DESCRIPTION
**Pull request by csmangum**
Six issues identified in review of the distillation pipeline — incorrect docstring, silent failure on empty input, inconsistent checkpoint state, always-on latency benchmarking, missing input validation, and a misleading CLI flag description.

## Changes

**`farm/core/decision/training/trainer_distill.py`**

- **KL docstring direction**: Corrected loss formula to `KLDiv(p_t ‖ p_s) * T²` — the previous `KLDiv(p_s ‖ p_t)` was backwards relative to the actual `kl_div(log_softmax(student), softmax(teacher), log_target=False)` call
- **Empty states guard in `train()`**: Raises `ValueError` immediately on `len(states) == 0` instead of silently running epochs with zero-loss metrics and writing a spurious checkpoint
- **Consistent no-validation best checkpoint**: Replaced "final weights + `min(train_losses)` proxy" with per-epoch tracking (`_best_train_loss`), so `best_epoch`, `best_val_loss`, and `best_state_dict` always refer to the same epoch
  ```python
  # before: best_state_dict = final weights, best_val_loss = min(all train losses) — inconsistent
  # after: per-epoch update when train_loss < _best_train_loss
  if train_loss < _best_train_loss:
      _best_train_loss = train_loss
      metrics.best_val_loss = train_loss
      metrics.best_epoch = epoch
      best_state_dict = {k: v.cpu().clone() for k, v in self.student.state_dict().items()}
  ```
- **Optional latency in `validate()`**: Skips `_measure_latency()` (110 forward passes) when `n_latency_repeats <= 0`, setting both latencies to `0.0` — avoids unexpectedly slow unit tests
- **Input validation in `validate()`**: Raises `ValueError` for empty `states` or non-positive `k_values`; uses `isinstance(k, (int, np.integer))` to handle numpy integer types

**`scripts/run_distillation.py`**

- **`--alpha` help text**: Corrected to "Blending weight for the soft distillation loss (`loss = alpha*soft + (1-alpha)*hard`)" — the previous text said "Hard-loss blend weight", which was the opposite of the implementation

### CODE REVIEW

**Comment by csmangum** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/Dooders/AgentFarm/pull/602#pullrequestreview-4071651282)
